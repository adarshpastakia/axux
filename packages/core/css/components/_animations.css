/**
 * AxUX React UI Framework with Tailwind CSS
 * @author    : Adarsh Pastakia
 * @version   : 4.0.0
 * @copyright : 2024
 * @license   : MIT
 */

/******************* animated circles *******************/
@layer components {
  @keyframes check-stroke {
    100% {
      stroke-dashoffset: 0;
    }
  }

  @keyframes check-scale {
    0%,
    100% {
      transform: none;
    }

    50% {
      transform: scale3d(1.1, 1.1, 1);
    }
  }

  @keyframes check-fill {
    100% {
      box-shadow: inset 0 0 0 1em currentcolor;
    }
  }

  @keyframes progress-animate {
    0% {
      background-position-x: 0;
    }

    100% {
      background-position-x: 1em;
    }
  }

  .check-animation {
    @apply inline-block w-[1.5em] h-[1.5em] rounded-[50%] stroke-white text-primary-base fill-transparent;

    stroke-width: 4px;
    stroke-miterlimit: 10;
    box-shadow: inset 0 0 0 currentcolor;
    animation:
      check-fill 0.4s ease-in-out 0.4s forwards,
      check-scale 0.3s ease-in-out 0.9s both;
  }

  .check-animation__circle {
    @apply stroke-current fill-transparent;

    stroke-width: 8px;
    stroke-dasharray: 166;
    stroke-dashoffset: 166;
    stroke-miterlimit: 10;
    animation: check-stroke 0.6s cubic-bezier(0.65, 0, 0.45, 1) forwards;
  }

  .check-animation__icon {
    @apply origin-center;

    stroke-dasharray: 48;
    stroke-dashoffset: 48;
    animation: check-stroke 0.3s cubic-bezier(0.65, 0, 0.45, 1) 0.8s forwards;
  }
}

/******************* loading bars *******************/
@layer components {
  @keyframes bar-animation {
    0% {
      @apply h-2;
    }

    60% {
      @apply h-2;
    }

    80% {
      @apply h-8;
    }

    100% {
      @apply h-2;
    }
  }

  .bar-animation {
    @apply inline-flex items-end h-12 text-white;

    & li {
      @apply list-none w-3 h-2 mx-px relative
        before:absolute before:inset-0 before:opacity-50 before:bg-current
        after:absolute after:inset-0 after:opacity-30 dark:opacity-100;

      &::before {
        background-color: color-mix(in lab, currentcolor 50%, var(--ax-gray));
      }

      &::after {
        box-shadow: 0 0 2px 1px var(--ax-black) inset;
      }

      animation: bar-animation 1.5s ease-in-out infinite;
    }

    & li:nth-child(1) {
      animation-delay: -1.2s;
    }

    & li:nth-child(2) {
      animation-delay: -1s;
    }

    & li:nth-child(3) {
      animation-delay: -0.8s;
    }

    & li:nth-child(4) {
      animation-delay: -0.6s;
    }
  }
}

/******************* loading spinner *******************/
@layer components {
  @keyframes spinner-animation {
    to {
      transform: rotate(360deg);
    }
  }

  .spinner-animation {
    @apply inline-block text-muted relative w-[1em] h-[1em] rounded-[50%] border-[0.125em] border-transparent border-t-current border-opacity-75 leading-none
    after:content-[""] after:absolute after:-inset-[0.125em] after:rounded-[50%] after:border-[0.125em] after:border-current after:opacity-30;

    animation: spinner-animation 1.8s cubic-bezier(0.65, 0.25, 0.45, 0) infinite;
  }
}

/******************* loading bar *******************/
@layer components {
  @keyframes loading-indicator {
    from {
      transform: translateX(calc(-100% * var(--multiplier)));
    }

    to {
      transform: translateX(calc(100% * var(--multiplier)));
    }
  }

  .loading-indicator {
    --multiplier: 1;

    @apply grid-area-[loader] relative;

    & > div {
      @apply absolute z-10 bg-base py-px inset-0 after:h-0.5 after:block after:bg-primary-base;

      height: calc(0.125rem + 2px);

      &::after {
        animation: loading-indicator 1s cubic-bezier(0.65, 0, 0.45, 1) infinite;
      }
    }
  }

  [dir="rtl"] .loading-indicator {
    --multiplier: -1;
  }
}

/******************* card content *******************/
@layer components {
  .card-animation {
    & > div {
      @apply max-w-lg text-muted;

      & figure {
        @apply block bg-gray-500/20 size-10;
      }

      & span {
        @apply block h-2 rounded bg-gradient-to-r from-gray-600/15 via-gray-600/10 to-gray-600/15;
      }
    }
  }
}

/******************* progress bar/circle *******************/
@layer components {
  [class^="progress-"] span {
    @apply flex w-full h-full items-center justify-center select-none font-bold;
  }

  .progress-circle {
    @apply inline-block w-[4em] h-[4em] text-lg relative;

    & > span {
      font-size: 0.625em;
    }

    & > svg {
      @apply absolute inset-0 w-full h-full;

      & circle {
        transition: stroke-dashoffset 0.5s;
      }
    }
  }

  .progress-bar {
    @apply block w-3/4 h-[1em] bg-dimmed mx-auto relative rounded font-bold overflow-hidden whitespace-nowrap;

    & > div:last-child {
      @apply h-full overflow-hidden bg-current relative;

      width: calc(var(--progress) * 1%);
      transition: width 0.5s;

      &::after {
        @apply absolute inset-0 text-white text-center z-0 align-top transition-none;

        width: calc((100 / var(--progress)) * 100%);
        content: attr(data-label);
        font-size: 0.5em;
        line-height: 0.5em;
        padding: 0.75em;
      }
    }

    &[data-animate="true"] > div:last-child {
      background-image: linear-gradient(
        -45deg,
        rgb(255 255 255 / 15%) 25%,
        transparent 25%,
        transparent 50%,
        rgb(255 255 255 / 15%) 50%,
        rgb(255 255 255 / 15%) 75%,
        transparent 75%,
        transparent
      );
      background-size: 1em 1em;
      animation: progress-animate 1s linear infinite;
    }

    & > div:first-child {
      @apply absolute inset-0 text-center z-0 align-top;

      font-size: 0.5em;
      line-height: 0.5em;
      padding: 0.75em;
    }
  }
}

/******************* progress colors/sizes *******************/
@layer utilities {
  [class^="progress-"][data-size="md"] {
    @apply text-2xl;
  }

  [class^="progress-"][data-size="lg"] {
    @apply text-3xl;
  }

  [class^="progress-"][data-size="xl"] {
    @apply text-4xl;
  }

  [class^="progress-"][data-color="primary"] {
    @apply text-primary-base;
  }

  [class^="progress-"][data-color="accent"] {
    @apply text-accent-base;
  }

  [class^="progress-"][data-color="danger"] {
    @apply text-danger-base;
  }

  [class^="progress-"][data-color="success"] {
    @apply text-success-base;
  }

  [class^="progress-"][data-color="warning"] {
    @apply text-warning-base;
  }
}
