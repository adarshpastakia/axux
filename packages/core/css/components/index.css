/**
 * AxUX React UI Framework with Pure CSS
 * @author    : Adarsh Pastakia
 * @version   : 4.0.0
 * @copyright : 2024
 * @license   : MIT
 */
