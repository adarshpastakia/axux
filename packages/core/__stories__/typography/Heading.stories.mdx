import {
  Anchor,
  ArgsTable,
  Canvas,
  Description,
  Heading,
  Meta,
  Story,
  Title
} from "@storybook/addon-docs";
import { AxHeading } from "../../src";
import { ColorStory, DisplayStory, HeadingStory, SerifStory } from "./HeadingStory";

<Meta
  title="@core/Typography/Heading"
  component={AxHeading}
  parameters={{ jest: ["Heading.text.tsx"] }}
/>

<Anchor storyId="core-typography-heading--heading">
  <Title>Heading</Title>
</Anchor>

<Description of={AxHeading} />

<Canvas>
  <Story name="Heading">{HeadingStory}</Story>
</Canvas>

<Anchor storyId="core-typography-heading--display-heading">
  <Heading>Display Heading</Heading>
</Anchor>

<Canvas>
  <Story name="Display Heading">{DisplayStory}</Story>
</Canvas>

<Anchor storyId="core-typography-heading--serif-font">
  <Heading>Serif Font</Heading>
</Anchor>

<Canvas>
  <Story name="Serif Font">{SerifStory}</Story>
</Canvas>

<Anchor storyId="core-typography-heading--color">
  <Heading>Color</Heading>
</Anchor>

<Canvas>
  <Story name="Color">{ColorStory}</Story>
</Canvas>

<Heading>Props</Heading>

<ArgsTable of={AxHeading} sort="requiredFirst" />
