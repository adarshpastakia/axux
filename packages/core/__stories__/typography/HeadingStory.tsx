// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2021
// @license   : MIT

import { Story } from "@storybook/react";
import { PropsWithChildren } from "react";
import { LIPSUM } from "../../../../storybook/components/Lipsum";
import { AxHeading } from "../../src";
import { HeadingProps } from "../../src/typography/Heading";

const Template: Story<PropsWithChildren<HeadingProps>> = (props) => <AxHeading {...props} />;

export const HeadingStory = Template.bind({});
HeadingStory.args = {
  children: LIPSUM.text,
  level: 1
};

export const DisplayStory = Template.bind({});
DisplayStory.args = {
  children: LIPSUM.text,
  level: 1,
  forDisplay: true
};

export const SerifStory = Template.bind({});
SerifStory.args = {
  children: LIPSUM.text,
  font: "serif"
};

export const ColorStory = Template.bind({});
ColorStory.args = {
  children: LIPSUM.text,
  bg: "lightest",
  color: "secondary",
  className: "ax-padding--sm"
};

export default { title: "Example/Heading", component: AxHeading };
