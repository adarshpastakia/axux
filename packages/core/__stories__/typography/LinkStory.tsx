// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2021
// @license   : MIT

import { Story } from "@storybook/react";
import { PropsWithChildren } from "react";
import { BrowserRouter } from "react-router-dom";
import { AxLink } from "../../src";
import { LinkProps } from "../../src/typography/Link";

const Template: Story<PropsWithChildren<LinkProps>> = (props) => (
  <BrowserRouter>
    <AxLink {...props} />
  </BrowserRouter>
);

export const LinkStory = Template.bind({});
LinkStory.args = {
  children: "Simple link",
  href: "#/test"
};

export default { title: "Example/Link", component: AxLink };
