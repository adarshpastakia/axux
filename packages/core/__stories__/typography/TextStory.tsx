// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2021
// @license   : MIT

import { Story } from "@storybook/react";
import { PropsWithChildren } from "react";
import { LIPSUM } from "../../../../storybook/components/Lipsum";
import { AxText } from "../../src";
import { TextProps } from "../../src/typography/Text";

const Template: Story<PropsWithChildren<TextProps>> = (props) => <AxText {...props} />;

export const TextStory = Template.bind({});
TextStory.args = {
  children: LIPSUM.para
};

export const SerifStory = Template.bind({});
SerifStory.args = {
  children: LIPSUM.para,
  className: "ax-text--dropcaps ax-align--justify",
  block: true,
  font: "serif"
};

export const ColorStory = Template.bind({});
ColorStory.args = {
  children: LIPSUM.para,
  className: "ax-text--indent-2 ax-align--justify ax-padding--sm",
  block: true,
  color: "secondary",
  bg: "lightest"
};

export default { title: "Example/Text", component: AxText };
