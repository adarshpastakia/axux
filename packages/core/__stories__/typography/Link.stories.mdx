import {
  Anchor,
  ArgsTable,
  Canvas,
  Description,
  Heading,
  Meta,
  Story,
  Title
} from "@storybook/addon-docs";
import { AxLink } from "../../src";
import { LinkStory } from "./LinkStory";

<Meta title="@core/typography/Link" component={AxLink} parameters={{ jest: ["Link.test.tsx"] }} />

<Anchor storyId="core-typography-link--link">
  <Title>Anchor Link</Title>
</Anchor>

<Description of={AxLink} />

<Canvas>
  <Story name="Link">{LinkStory}</Story>
</Canvas>

<Heading>Props</Heading>

<ArgsTable of={AxLink} sort="requiredFirst" />
