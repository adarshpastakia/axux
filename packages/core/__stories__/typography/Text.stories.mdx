import {
  Anchor,
  ArgsTable,
  Canvas,
  Description,
  Heading,
  Meta,
  Story,
  Title
} from "@storybook/addon-docs";
import { AxFlexBox, AxText } from "../../src";
import { ColorStory, SerifStory, TextStory } from "./TextStory";
import { LIPSUM } from "../../../../storybook/components/Lipsum";

<Meta title="@core/Typography/Text" component={AxText} parameters={{ jest: ["Text.test.tsx"] }} />

<Anchor storyId="core-typography-text--text">
  <Title>Text</Title>
</Anchor>

<Description of={AxText} />

<Canvas>
  <Story name="Text">{TextStory}</Story>
</Canvas>

<Anchor storyId="core-typography-text--serif-font">
  <Heading>Serif Font</Heading>
</Anchor>

<Canvas>
  <Story name="Serif Font">{SerifStory}</Story>
</Canvas>

<Anchor storyId="core-typography-text--color">
  <Heading>Color</Heading>
</Anchor>

<Canvas>
  <Story name="Color">{ColorStory}</Story>
</Canvas>

<Anchor storyId="core-typography-text--weights">
  <Heading>Weights</Heading>
</Anchor>

<Canvas columns={6}>
  <Story name="Weights">
    <AxFlexBox.Row>
      <AxText weight="black">Black text</AxText>
      <AxText weight="bold">Bold text</AxText>
      <AxText weight="medium">Medium text</AxText>
      <AxText weight="normal">Normal text</AxText>
      <AxText weight="light">Light text</AxText>
      <AxText weight="thin">Thin text</AxText>
    </AxFlexBox.Row>
  </Story>
</Canvas>

<Anchor storyId="core-typography-text--sizes">
  <Heading>Sizes</Heading>
</Anchor>

<Canvas columns={6}>
  <Story name="Sizes">
    <AxFlexBox.Row align="middle">
      <AxText size="xs">XSmall text</AxText>
      <AxText size="sm">Small text</AxText>
      <AxText size="default">Default text</AxText>
      <AxText size="md">Medium text</AxText>
      <AxText size="lg">Large text</AxText>
      <AxText size="xl">XLarge text</AxText>
      <AxText className="ax-col--full" size="2em">
        2em text
      </AxText>
    </AxFlexBox.Row>
  </Story>
</Canvas>

<Anchor storyId="core-typography-text--clipped">
  <Heading>Clipped</Heading>
</Anchor>

<Canvas>
  <Story name="Clipped">
    <div style={{ maxWidth: 1200 }}>
      <AxText clip={7}>
        <p>{LIPSUM.para}</p>
        <p>{LIPSUM.para}</p>
      </AxText>
    </div>
  </Story>
</Canvas>

<Anchor storyId="core-typography-text--marked-text">
  <Heading>Marked text</Heading>
</Anchor>

<Canvas>
  <Story name="Marked text">
    <AxText mark={LIPSUM.text}>{LIPSUM.para}</AxText>
  </Story>
</Canvas>

<Anchor storyId="core-typography-text--tooltip-text">
  <Heading>Tooltip text</Heading>
</Anchor>

<Canvas>
  <Story name="Tooltip text">
    <AxText tooltip="Test tooltip">{LIPSUM.text}</AxText>
    <AxText>{LIPSUM.para.replace(LIPSUM.text, "")}</AxText>
  </Story>
</Canvas>

<Heading>Props</Heading>

<ArgsTable of={AxText} sort="requiredFirst" />
