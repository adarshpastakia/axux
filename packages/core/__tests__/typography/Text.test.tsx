// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2021
// @license   : MIT

import { composeStories } from "@storybook/testing-react";
import { act, fireEvent, render, waitFor } from "@testing-library/react";
import i18next from "i18next";
import { I18nextProvider, initReactI18next } from "react-i18next";
import { LIPSUM } from "../../../../storybook/components/Lipsum";
import * as stories from "../../__stories__/typography/TextStory";

const { TextStory, SerifStory, ColorStory } = composeStories(stories);

i18next
  .use(initReactI18next)
  .init({
    resources: {
      en: {}
    },
    defaultNS: "core",
    fallbackLng: ["en"],
    keySeparator: ".",

    interpolation: {
      escapeValue: false // not needed for react as it escapes by default
    }
  })
  .then();

describe("Text", () => {
  it("renders simple text", (done) => {
    const fragment = render(
      <I18nextProvider i18n={i18next}>
        <TextStory />
      </I18nextProvider>
    );
    expect(fragment.container.querySelector(".ax-text")).toBeInTheDocument();
    expect(fragment.container).toMatchSnapshot();
    fragment.unmount();
    done();
  });

  it("renders serif text", (done) => {
    const fragment = render(
      <I18nextProvider i18n={i18next}>
        <SerifStory />
      </I18nextProvider>
    );
    expect(fragment.container.querySelector(".ax-font--serif")).toBeInTheDocument();
    expect(fragment.container).toMatchSnapshot();
    fragment.unmount();
    done();
  });

  it("renders color text", (done) => {
    const fragment = render(
      <I18nextProvider i18n={i18next}>
        <ColorStory />
      </I18nextProvider>
    );
    expect(fragment.container.querySelector(".ax-bg--lightest")).toBeInTheDocument();
    expect(fragment.container.querySelector(".ax-color--secondary")).toBeInTheDocument();
    expect(fragment.container).toMatchSnapshot();
    fragment.unmount();
    done();
  });

  it("renders weighted text", (done) => {
    const fragment = render(
      <I18nextProvider i18n={i18next}>
        <TextStory weight="bold" />
      </I18nextProvider>
    );
    expect(fragment.container.querySelector(".ax-weight--bold")).toBeInTheDocument();
    expect(fragment.container).toMatchSnapshot();
    fragment.unmount();
    done();
  });

  it("renders sized text", (done) => {
    const fragment = render(
      <I18nextProvider i18n={i18next}>
        <TextStory size="md" />
        <TextStory size="10px" />
        <TextStory size={3} />
      </I18nextProvider>
    );
    expect(fragment.container.querySelector(".ax-font--md")).toBeInTheDocument();
    expect(fragment.container).toMatchSnapshot();
    fragment.unmount();
    done();
  });

  it("renders block text", (done) => {
    const fragment = render(
      <I18nextProvider i18n={i18next}>
        <TextStory block />
      </I18nextProvider>
    );
    expect(fragment.container.querySelector(".ax-block")).toBeInTheDocument();
    expect(fragment.container).toMatchSnapshot();
    fragment.unmount();
    done();
  });

  it("renders clipped text", (done) => {
    const fragment = render(
      <I18nextProvider i18n={i18next}>
        <TextStory clip={2} />
      </I18nextProvider>
    );
    expect(fragment.container.querySelector(".ax-text")).toBeInTheDocument();
    expect(fragment.container).toMatchSnapshot();
    fragment.unmount();
    done();
  });

  it("renders marked text", (done) => {
    const fragment = render(
      <I18nextProvider i18n={i18next}>
        <TextStory mark={LIPSUM.text} />
      </I18nextProvider>
    );
    expect(fragment.container.querySelector("mark")).toBeInTheDocument();
    expect(fragment.container).toMatchSnapshot();
    fragment.unmount();
    done();
  });

  it("renders marked text when not found", (done) => {
    let fragment = render(
      <I18nextProvider i18n={i18next}>
        <TextStory mark="Test me" />
      </I18nextProvider>
    );
    expect(fragment.container.querySelector("mark")).toBeNull();
    expect(fragment.container).toMatchSnapshot();
    fragment.unmount();

    fragment = render(
      <I18nextProvider i18n={i18next}>
        <TextStory mark={LIPSUM.text}>
          <b>{LIPSUM.line}</b>
        </TextStory>
      </I18nextProvider>
    );
    expect(fragment.container.querySelector("mark")).toBeNull();
    expect(fragment.container).toMatchSnapshot();
    fragment.unmount();
    done();
  });

  it("renders with tooltip", async () => {
    const fragment = render(
      <I18nextProvider i18n={i18next}>
        <div className="ax-root">
          <TextStory abbr={[["IPSUM", "Test tooltip"]]} />
        </div>
      </I18nextProvider>
    );

    const el = fragment.container.querySelector("abbr") as HTMLElement;
    act(() => {
      fireEvent.mouseOver(el);
    });
    await waitFor(() => fragment.container.querySelector(".ax-tooltip"));
    expect(fragment.container).toHaveTextContent("Test tooltip");
    expect(fragment.container).toMatchSnapshot();
    act(() => {
      fireEvent.mouseOut(el);
    });
    expect(fragment.container).not.toHaveTextContent("Test tooltip");
    expect(fragment.container).toMatchSnapshot();
    fragment.unmount();
  });
});
