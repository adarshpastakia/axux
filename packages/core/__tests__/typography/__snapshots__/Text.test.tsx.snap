// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders block text 1`] = `
<div>
  <span
    class="ax-text ax-block  "
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent quis eros mollis,
          egestas mauris nec, tincidunt ipsum. Sed commodo sagittis sapien non sagittis. Vestibulum
          viverra sapien vitae nisi condimentum, eu malesuada velit iaculis. Duis pulvinar lobortis
          diam, a pharetra urna commodo non. Integer molestie pharetra enim a pulvinar. Fusce
          ultrices augue vel magna tincidunt, auctor efficitur lectus gravida. Nunc scelerisque urna
          nec augue tincidunt, ac aliquet velit facilisis.
  </span>
</div>
`;

exports[`Text renders clipped text 1`] = `
<div>
  <span
    class="ax-text ax-block ax-align--justify"
    style="--line-clamp: 2; display: block;"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent quis eros mollis,
          egestas mauris nec, tincidunt ipsum. Sed commodo sagittis sapien non sagittis. Vestibulum
          viverra sapien vitae nisi condimentum, eu malesuada velit iaculis. Duis pulvinar lobortis
          diam, a pharetra urna commodo non. Integer molestie pharetra enim a pulvinar. Fusce
          ultrices augue vel magna tincidunt, auctor efficitur lectus gravida. Nunc scelerisque urna
          nec augue tincidunt, ac aliquet velit facilisis.
  </span>
</div>
`;

exports[`Text renders color text 1`] = `
<div>
  <span
    class="ax-text ax-block ax-text--indent-2 ax-align--justify ax-padding--sm ax-bg--lightest ax-color--secondary "
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent quis eros mollis,
          egestas mauris nec, tincidunt ipsum. Sed commodo sagittis sapien non sagittis. Vestibulum
          viverra sapien vitae nisi condimentum, eu malesuada velit iaculis. Duis pulvinar lobortis
          diam, a pharetra urna commodo non. Integer molestie pharetra enim a pulvinar. Fusce
          ultrices augue vel magna tincidunt, auctor efficitur lectus gravida. Nunc scelerisque urna
          nec augue tincidunt, ac aliquet velit facilisis.
  </span>
</div>
`;

exports[`Text renders marked text 1`] = `
<div>
  <span
    class="ax-text   "
  >
    <mark>
      Lorem ipsum
    </mark>
    <span>
       dolor sit amet, consectetur adipiscing elit. Praesent quis eros mollis,
          egestas mauris nec, tincidunt ipsum. Sed commodo sagittis sapien non sagittis. Vestibulum
          viverra sapien vitae nisi condimentum, eu malesuada velit iaculis. Duis pulvinar lobortis
          diam, a pharetra urna commodo non. Integer molestie pharetra enim a pulvinar. Fusce
          ultrices augue vel magna tincidunt, auctor efficitur lectus gravida. Nunc scelerisque urna
          nec augue tincidunt, ac aliquet velit facilisis.
    </span>
  </span>
</div>
`;

exports[`Text renders marked text when not found 1`] = `
<div>
  <span
    class="ax-text   "
  >
    <span>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent quis eros mollis,
          egestas mauris nec, tincidunt ipsum. Sed commodo sagittis sapien non sagittis. Vestibulum
          viverra sapien vitae nisi condimentum, eu malesuada velit iaculis. Duis pulvinar lobortis
          diam, a pharetra urna commodo non. Integer molestie pharetra enim a pulvinar. Fusce
          ultrices augue vel magna tincidunt, auctor efficitur lectus gravida. Nunc scelerisque urna
          nec augue tincidunt, ac aliquet velit facilisis.
    </span>
  </span>
</div>
`;

exports[`Text renders marked text when not found 2`] = `
<div>
  <span
    class="ax-text   "
  >
    <b>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </b>
  </span>
</div>
`;

exports[`Text renders serif text 1`] = `
<div>
  <span
    class="ax-text ax-block ax-text--dropcaps ax-align--justify ax-font--serif "
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent quis eros mollis,
          egestas mauris nec, tincidunt ipsum. Sed commodo sagittis sapien non sagittis. Vestibulum
          viverra sapien vitae nisi condimentum, eu malesuada velit iaculis. Duis pulvinar lobortis
          diam, a pharetra urna commodo non. Integer molestie pharetra enim a pulvinar. Fusce
          ultrices augue vel magna tincidunt, auctor efficitur lectus gravida. Nunc scelerisque urna
          nec augue tincidunt, ac aliquet velit facilisis.
  </span>
</div>
`;

exports[`Text renders simple text 1`] = `
<div>
  <span
    class="ax-text   "
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent quis eros mollis,
          egestas mauris nec, tincidunt ipsum. Sed commodo sagittis sapien non sagittis. Vestibulum
          viverra sapien vitae nisi condimentum, eu malesuada velit iaculis. Duis pulvinar lobortis
          diam, a pharetra urna commodo non. Integer molestie pharetra enim a pulvinar. Fusce
          ultrices augue vel magna tincidunt, auctor efficitur lectus gravida. Nunc scelerisque urna
          nec augue tincidunt, ac aliquet velit facilisis.
  </span>
</div>
`;

exports[`Text renders sized text 1`] = `
<div>
  <span
    class="ax-text   ax-font--md "
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent quis eros mollis,
          egestas mauris nec, tincidunt ipsum. Sed commodo sagittis sapien non sagittis. Vestibulum
          viverra sapien vitae nisi condimentum, eu malesuada velit iaculis. Duis pulvinar lobortis
          diam, a pharetra urna commodo non. Integer molestie pharetra enim a pulvinar. Fusce
          ultrices augue vel magna tincidunt, auctor efficitur lectus gravida. Nunc scelerisque urna
          nec augue tincidunt, ac aliquet velit facilisis.
  </span>
  <span
    class="ax-text   "
    style="font-size: 10px;"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent quis eros mollis,
          egestas mauris nec, tincidunt ipsum. Sed commodo sagittis sapien non sagittis. Vestibulum
          viverra sapien vitae nisi condimentum, eu malesuada velit iaculis. Duis pulvinar lobortis
          diam, a pharetra urna commodo non. Integer molestie pharetra enim a pulvinar. Fusce
          ultrices augue vel magna tincidunt, auctor efficitur lectus gravida. Nunc scelerisque urna
          nec augue tincidunt, ac aliquet velit facilisis.
  </span>
  <span
    class="ax-text   "
    style="font-size: 3rem;"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent quis eros mollis,
          egestas mauris nec, tincidunt ipsum. Sed commodo sagittis sapien non sagittis. Vestibulum
          viverra sapien vitae nisi condimentum, eu malesuada velit iaculis. Duis pulvinar lobortis
          diam, a pharetra urna commodo non. Integer molestie pharetra enim a pulvinar. Fusce
          ultrices augue vel magna tincidunt, auctor efficitur lectus gravida. Nunc scelerisque urna
          nec augue tincidunt, ac aliquet velit facilisis.
  </span>
</div>
`;

exports[`Text renders weighted text 1`] = `
<div>
  <span
    class="ax-text   ax-weight--bold "
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent quis eros mollis,
          egestas mauris nec, tincidunt ipsum. Sed commodo sagittis sapien non sagittis. Vestibulum
          viverra sapien vitae nisi condimentum, eu malesuada velit iaculis. Duis pulvinar lobortis
          diam, a pharetra urna commodo non. Integer molestie pharetra enim a pulvinar. Fusce
          ultrices augue vel magna tincidunt, auctor efficitur lectus gravida. Nunc scelerisque urna
          nec augue tincidunt, ac aliquet velit facilisis.
  </span>
</div>
`;

exports[`Text renders with tooltip 1`] = `
<div>
  <div
    class="ax-root"
  >
    <span
      class="ax-text   "
    >
      <span>
        Lorem 
      </span>
      <abbr
        class="ax-color--"
      >
        ipsum
      </abbr>
      <span>
         dolor sit amet, consectetur adipiscing elit. Praesent quis eros mollis,
          egestas mauris nec, tincidunt 
      </span>
      <abbr
        class="ax-color--"
      >
        ipsum
      </abbr>
      <span>
        . Sed commodo sagittis sapien non sagittis. Vestibulum
          viverra sapien vitae nisi condimentum, eu malesuada velit iaculis. Duis pulvinar lobortis
          diam, a pharetra urna commodo non. Integer molestie pharetra enim a pulvinar. Fusce
          ultrices augue vel magna tincidunt, auctor efficitur lectus gravida. Nunc scelerisque urna
          nec augue tincidunt, ac aliquet velit facilisis.
      </span>
    </span>
    <div
      class="ax-popper ax-tooltip ax-bg--darkest ax-color--empty "
      data-popper-escaped="true"
      data-popper-placement="top"
      data-popper-reference-hidden="true"
      data-show="true"
      data-show-arrow="true"
      style="position: fixed; left: 0px; top: 0px; min-width: 0; bottom: 0px; transform: translate(0px, -10px);"
    >
      <div
        class="ax-popper--body"
      >
        <span>
          Test tooltip
        </span>
      </div>
      <div
        class="ax-popper--arrow"
        style="position: absolute; left: 0px; transform: translate(0px, 0px);"
      />
    </div>
  </div>
</div>
`;

exports[`Text renders with tooltip 2`] = `
<div>
  <div
    class="ax-root"
  >
    <span
      class="ax-text   "
    >
      <span>
        Lorem 
      </span>
      <abbr
        class="ax-color--"
      >
        ipsum
      </abbr>
      <span>
         dolor sit amet, consectetur adipiscing elit. Praesent quis eros mollis,
          egestas mauris nec, tincidunt 
      </span>
      <abbr
        class="ax-color--"
      >
        ipsum
      </abbr>
      <span>
        . Sed commodo sagittis sapien non sagittis. Vestibulum
          viverra sapien vitae nisi condimentum, eu malesuada velit iaculis. Duis pulvinar lobortis
          diam, a pharetra urna commodo non. Integer molestie pharetra enim a pulvinar. Fusce
          ultrices augue vel magna tincidunt, auctor efficitur lectus gravida. Nunc scelerisque urna
          nec augue tincidunt, ac aliquet velit facilisis.
      </span>
    </span>
  </div>
</div>
`;
